cmds/cpio/cpio.go

```diff
@@ -83,11 +83,12 @@ func main() {
 
 	case "o":
 		rw := archiver.Writer(os.Stdout)
```

Comment:
not having looked this much, but could you embed the Recorder in the archiver.Writer? Then you would not need to add a new parameter everywhere? It could be I'm not looking correctly at your solution however. But still the new added parameter is a bit concerning.

Reply:
Happy to do those changes if somebody can tell me that's what they want for the interfaces with high certainty. I certainly don't know what the goal is, and I wouldn't want to stomp on somebody's idea. As discussed on slack I believe this is the smallest change that will work, so let's clean it up later if we find that we want that.